src/coreclr/src/jit/morph.cpp

```diff
@@ -12899,13 +12899,15 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                 }
                 else if (tree->IsUnsigned() && op2->IsIntegralConst(0))
                 {
-                    if ((oper == GT_GT) || (oper == GT_LE))
+                    if (!op1->OperIs(GT_ARR_LENGTH) && ((oper == GT_GT) || (oper == GT_LE)))
                     {
                         // IL doesn't have a cne instruction so compilers use cgt.un instead. The JIT
                         // recognizes certain patterns that involve GT_NE (e.g (x & 4) != 0) and fails
                         // if GT_GT is used instead. Transform (x GT_GT.unsigned 0) into (x GT_NE 0)
                         // and (x GT_LE.unsigned 0) into (x GT_EQ 0). The later case is rare, it sometimes
                         // occurs as a result of branch inversion.
+                        // NOTE: keep it as it is for Array.Length op1 since GT_NE/GT_EQ operators are
+                        // not bounds check elimination friendly yet.
```

Comment:
Do you know how hard it is to make it friendly?
As I see we use https://github.com/dotnet/runtime/blob/8527a991b91cabc9bec6f6abe57a52b5bb5e154d/src/coreclr/src/jit/gentree.h#L1563 to find what we optimize as a bound check and we don't have `EQ/NE` there, should we add them?